packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationRegistry.m

```diff
@@ -214,6 +230,10 @@ - (void)moduleDidForeground:(id)module
 
 - (void)moduleDidBackground:(id)module
 {
+  // We save the mask to restore it when the app moves to the foreground.
+  // We don't want to wait for the module to call moduleDidForeground, cause it will add unnecessary rotation.
+  _lastOrientationMask = [self requiredOrientationMask];
```

Comment:
Should we call this only if `_foregroundedModule == module`?

Reply:
Nope, cause module can change mask when the app is running. So, the best place for this code is here in my option :D

Reply:
> module can change mask when the app is running

Sure, but we will only ever fetch a value from this variable if `_foregroundedModule` is `nil`, which only happens if:
- no module has been foregrounded yet (from the start), we return `0` in that case, `_lastOrientationMask` isn't taken into account,
- a module has been backgrounded, which can only happen if 
https://github.com/expo/expo/blob/ee29508092dd1e94e343cbe5ecdd1d45c6fcc1ac/packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationRegistry.m#L217-L218, so what I mean is that we also could just set the value once we go into the state in which we will read it.